src/daft-core/src/array/ops/hll_sketch.rs

```diff
@@ -0,0 +1,69 @@
+use crate::{
+    array::ops::as_arrow::AsArrow,
+    datatypes::{FixedSizeBinaryArray, UInt64Array},
+    DataType,
+};
+use hyperloglog::{HyperLogLog, NUM_REGISTERS};
+
+use common_error::DaftResult;
+
+use crate::array::ops::{DaftHllSketchAggable, GroupIndices};
+
+pub const HLL_SKETCH_DTYPE: DataType = DataType::FixedSizeBinary(NUM_REGISTERS);
+
+impl DaftHllSketchAggable for UInt64Array {
+    type Output = DaftResult<FixedSizeBinaryArray>;
+
+    fn hll_sketch(&self) -> Self::Output {
+        let mut hll = HyperLogLog::default();
+        match self.validity() {
+            Some(validity) => {
+                for (index, &value) in self.as_arrow().values_iter().enumerate() {
+                    if validity.get_bit(index) {
+                        hll.add_already_hashed(value);
+                    };
+                }
+            }
+            None => {
+                for &value in self.as_arrow().values_iter() {
+                    hll.add_already_hashed(value);
+                }
+            }
+        }
+        let array = (self.name(), hll.registers.as_ref() as &[u8]).into();
```

Comment:
You could skip a copy here by returning `hll.registers` as a `Vec<u8>`

Reply:
Still need to perform a copy since `hll.registers` is `Cow<'_, [u8]>`. I have to perform a `hll.registers.to_vec()`.

However, changing the `impl From<(.., &[u8])> for FixedSizeBinaryArray` to `impl From<(.., Vec<u8>)> for FixedSizeBinaryArray` was good because the end-user is in control of the memcpy, not the API.

This helped later in the `grouped_hll_sketch`, I believe.